### PR TITLE
prov/rxd: Add rxd_remove_free_pkt_entry for dlist_remove/buf_free of pkt_entries

### PR DESCRIPTION
**Pull request by nikitaxgusev**
Signed-off-by: Nikita Gusev <nikita.gusev@intel.com>

### CODE REVIEW

**Comment by nikitaxgusev** [comment]
@a-ilango  I think, the build was failed because of infrastructure issues, wasn't  it?

**Comment by a-ilango** [comment]
> @a-ilango I think, the build was failed because of infrastructure issues, wasn't it?

Yes, travis CI ran into some issue. I've restarted the build.

**Comment by nikitaxgusev** [comment]
@a-ilango  Thanks.

File: prov/rxd/src/rxd.h [hunk 1]
```diff
@@ -364,6 +364,12 @@ static inline void *rxd_pkt_start(struct rxd_pkt_entry *pkt_entry)
 	return (void *) ((char *) pkt_entry + sizeof(*pkt_entry));
 }
 
+static inline void rxd_dequeued_pkt_entry(struct rxd_pkt_entry *pkt_entry)
```
Comment (aingerson) [inline prov/rxd/src/rxd.h]: I don't love the name of the new function. Can we rename it to something like
rxd_remove_free_pkt_entry ?
I want to make it clear that it is also freeing the buffer.

Comment (nikitaxgusev) [reply prov/rxd/src/rxd.h]: @aingerson Thanks. Done.
Renamed the function to `rxd_remove_free_pkt_entry`.

**Pull request merged**